api/opds2.py

```diff
@@ -58,3 +79,129 @@ def feed_links(self):
         return [
             {"href": self.url, "rel": "self", "type": self.OPDS2_TYPE},
         ]
+
+
+class OPDS2API(BaseCirculationAPI):
+    """
+    Generic OPDS2 Circulation APIs
+    This should be a pure implementation with no references to daatasources
+    """
+
+    def __init__(self, db: Session, collection: Collection) -> None:
+        self._db = db
+        self.collection = collection
+        self.log = logging.getLogger("OPDS2API")
+        self.token_auth_url = self._get_setting(ExternalIntegration.TOKEN_AUTH)
+
+    def _get_setting(self, key) -> str | None:
+        return ConfigurationSetting.for_externalintegration(
+            key, self.collection.external_integration
+        ).value
+
+    def _get_authentication_token(self, patron: Patron) -> ProblemDetail | str:
+        """Get the authentication token for a patron"""
+        url = URITemplate(self.token_auth_url).expand(patron_id=patron.username)
+        response = HTTP.get_with_timeout(url)
+        if response.status_code != 200:
+            self.log.error(
+                f"Could not authenticate the patron({patron.username}): {response.content}"
+            )
+            return INVALID_CREDENTIALS
+
+        # The response should be the JWT token, not wrapped in any format like JSON
+        token = response.json().get("token")
```

Comment:
The comment and the logic here do not match. The comment is correct though, the token should be just plain text that we can directly template into the link.

Reply:
The plain text token matches what we saw in the actual feed.